StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1117ParametersMustBeOnSameLineOrSeparateLines.cs

```diff
@@ -55,24 +59,319 @@ public class SA1117ParametersMustBeOnSameLineOrSeparateLines : DiagnosticAnalyze
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleAnonymousMethod, SyntaxKind.AnonymousMethodExpression);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleLambda, SyntaxKind.ParenthesizedLambdaExpression);
+        }
+
+        private static void HandleBaseMethodDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var declaration = (BaseMethodDeclarationSyntax)context.Node;
+            HandleParameterListSyntax(context, declaration.ParameterList);
+        }
+
+        private static void HandleInvocation(SyntaxNodeAnalysisContext context)
         {
-            get
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            HandleArgumentListSyntax(context, invocation.ArgumentList);
+        }
+
+        private static void HandleObjectCreation(SyntaxNodeAnalysisContext context)
+        {
+            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            HandleArgumentListSyntax(context, objectCreation.ArgumentList);
+        }
+
+        private static void HandleIndexerDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var indexerDeclaration = (IndexerDeclarationSyntax)context.Node;
+            BracketedParameterListSyntax argumentListSyntax = indexerDeclaration.ParameterList;
+            SeparatedSyntaxList<ParameterSyntax> arguments = argumentListSyntax.Parameters;
+
+            if (arguments.Count > 2)
             {
-                return SupportedDiagnosticsValue;
+                Analyze(context, argumentListSyntax.Parameters);
             }
         }
 
-        /// <inheritdoc/>
-        public override void Initialize(AnalysisContext context)
+        private static void HandleElementAccess(SyntaxNodeAnalysisContext context)
+        {
+            var elementAccess = (ElementAccessExpressionSyntax)context.Node;
+            HandleBracketedArgumentListSyntax(context, elementAccess.ArgumentList);
+        }
+
+        private static void HandleArrayCreation(SyntaxNodeAnalysisContext context)
+        {
+            var arrayCreation = (ArrayCreationExpressionSyntax)context.Node;
+
+            foreach (var rankSpecifier in arrayCreation.Type.RankSpecifiers)
+            {
+                SeparatedSyntaxList<ExpressionSyntax> sizes = rankSpecifier.Sizes;
+                if (sizes.Count < 3)
+                {
+                    continue;
+                }
+
+                ExpressionSyntax previousParameter = sizes[1];
+                int firstParameterLine = sizes[0].GetLine();
+                int previousLine = previousParameter.GetLine();
+                Func<int, int, bool> lineCondition;
+
+                if (firstParameterLine == previousLine)
+                {
+                    // arguments must be on same line
+                    lineCondition = (param1Line, param2Line) => param1Line != param2Line;
```

Comment:
For readability purposes I would suggest to invert the `lineCondition`. That way the code will match the comment made above it.